src/main.rs

```diff
@@ -29,8 +30,10 @@ fn main() {
 
 fn real_main() -> Result<i32, Box<dyn std::error::Error>> {
     let rng = Rng::new();
-    let cargo = std::env::var("CARGO").unwrap_or_else(|_| "cargo".to_owned());
-    let mut arg_iter = std::env::args().peekable();
+    let cargo = env::var("CARGO_MOMMY_ACTUAL")
+        .or_else(|_| env::var("CARGO"))
+        .unwrap_or_else(|_| "cargo".to_owned());
```

Comment:
If possible, i would like this to be `CARGO_{true_role}S_ACTUAL` (I *think* it should be, since you aren't using it in the computation of true_role?)